data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/tracing/TracingConfig.java

```diff
@@ -39,70 +69,124 @@ private TracingConfig(final Backend backend, final String url, final float sampl
 
     this.backend = backend;
     this.url = url;
-    this.samplingRate = Math.min(1, Math.max(samplingRate, 0));
+    if (backend.equals(Backend.UNKNOWN)) {
+      this.samplingRate = 0F;
+    } else {
+      this.samplingRate = Math.min(1, Math.max(samplingRate, 0));
+    }
   }
 
-  public static TracingConfig fromDir(final String path) throws IOException {
-
-    final var backendPath = backendPath(path);
-    if (!Files.exists(backendPath)) {
-      return TracingConfig.builder()
-        .setBackend(Backend.UNKNOWN)
-        .createTracingConfig();
+  public OpenTelemetrySdk setup() {
+    logger.info(
+      "Registering tracing configurations {} {} {} {}",
+      keyValue("backend", getBackend()),
+      keyValue("sampleRate", getSamplingRate()),
+      keyValue("url", getUrl()),
```

Comment:
`url` might contain sensitive data, keys or passwords of a monitoring provider, even if config-tracing by default is a CM, it's potentially possible to convert and use a secret but we should print it here.

Reply:
I don't recall this was there before this cleanup or not TBH, but I'll pr it